Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/BXLInvocationAnalyzer.cs

```diff
@@ -0,0 +1,95 @@
+
+using System;
+using System.IO;
+using BuildXL.Analyzers.Core.XLGPlusPlus;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.ToolSupport;
+using BuildXL.Utilities;
+using Google.Protobuf;
+
+namespace BuildXL.Execution.Analyzer
+{
+
+    internal partial class Args
+    {
+        public Analyzer InitializeDominoInvocationAnalyzer()
+        {
+            string inputDirPath = null;
+            string outputFilePath = null;
+
+            foreach (var opt in AnalyzerOptions)
+            {
+                if (opt.Name.Equals("outputFile", StringComparison.OrdinalIgnoreCase) ||
+                   opt.Name.Equals("o", StringComparison.OrdinalIgnoreCase))
+                {
+                    outputFilePath = ParseSingletonPathOption(opt, outputFilePath);
+                }
+                else if (opt.Name.Equals("inputDir", StringComparison.OrdinalIgnoreCase) ||
+                     opt.Name.Equals("i", StringComparison.OrdinalIgnoreCase))
+                {
+                    inputDirPath = ParseSingletonPathOption(opt, inputDirPath);
+                }
+                else
+                {
+                    throw Error("Unknown option for event stats analysis: {0}", opt.Name);
+                }
+            }
+
+
+            if (string.IsNullOrEmpty(inputDirPath))
+            {
+                throw Error("/inputDir parameter is required");
+            }
+
+            return new BXLInvocationAnalyzer(GetAnalysisInput())
+            {
+                InputDirPath = inputDirPath,
+                OutputFilePath = outputFilePath
+            };
+        }
+
+        /// <summary>
+        /// Write the help message when the analyzer is invoked with the /help flag
+        /// </summary>
+        private static void WriteDominoInvocationHelp(HelpWriter writer)
+        {
+            writer.WriteBanner("Domino Invocation \"Analyzer\"");
+            writer.WriteModeOption(nameof(AnalysisMode.BXLInvocationXLG), "Gets and outputs information related to domino invocation events from the database.");
+            writer.WriteOption("inputDir", "Required. The directory to read the RocksDB database from", shortName: "i");
+            writer.WriteOption("outputFile", "Required. The file where to write the results", shortName: "o");
+        }
+    }
+
+    /// <summary>
+    /// Analyzer to examine BXL Invocation events that have been dumped into the db
+    /// </summary>
+    internal sealed class BXLInvocationAnalyzer : Analyzer
+    {
+
+        public string InputDirPath;
+        public string OutputFilePath;
+
+        public BXLInvocationAnalyzer(AnalysisInput input): base(input)
+        {
+
+        }
+
+        /// <inheritdoc/>
+        public override int Analyze()
+        {
+            var dataStore = new XLGppDataStore();
+            if (dataStore.OpenDatastore(storeDirectory: InputDirPath))
+            {
+                dataStore.GetEventsByType((int)ExecutionEventId.DominoInvocation);

```

Comment:
>dataStore.GetEventsByType((int)ExecutionEventId.DominoInvocation); [](start = 16, length = 66)

so this currently returns a list but you are not doing anything with it... shouldn't it be stored/saved under `OutputFilePath`?

Reply:
Yes eventually I would want to but I was a bit unsure how useful that would be for now since its just one event, but as I add more events I would remove the Console writes and make it output to a file :) 

---
In reply to: [306053507](https://github.com/microsoft/BuildXL/pull/636#discussion_r306053507) [](ancestors = 306053507)